src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -124,18 +124,9 @@ class CoreWorkerDirectActorTaskSubmitter {
 
   /// Map from actor id to the actor's pending requests. Each actor's requests
   /// are ordered by the task number in the request.
-  absl::flat_hash_map<ActorID, std::map<int64_t, std::unique_ptr<rpc::PushTaskRequest>>>
+  absl::flat_hash_map<ActorID, std::vector<std::unique_ptr<rpc::PushTaskRequest>>>
```

Comment:
Hmm why the change to a vector? Wouldn't this cause the vector to no longer be ordered in some cases?

Reply:
It does, but there's no reason why the messages need to be submitted in a fixed order. Basically what was happening before was that requests were ordered in both the client and the server. Sorting in the client is unnecessary (gRPC doesn't retain it anyway), so I removed it. And of course this is valid only because sequence numbers are already fixed when `TaskSpecification`s are created (and later only propagated to `PushTaskRequest`s).

Reply:
I think we needed to do this on the client end too because the receiving end will time out if it receives an out-of-order request and doesn't receive the earlier request in time. Usually it's okay, but there are cases where consecutively submitted tasks will get sent at significantly different times because we don't send until the object dependencies have been resolved. @ericl can confirm.

Reply:
Yes, I've been concerned about that too, but I've decided that in the end the timeout solution is pretty arbitrary too. Ordering client-side can help a bit, but it seems functionally equivalent to allowing longer wait times (unless the distribution of dependency resolution has an extremely long tail).

In fact, why don't we remove that timeout? It was necessary before because we could never retry the tasks, but the current implementation will keep resubmitting any failed requests indefinitely, so it should not be necessary anymore.

Reply:
Removing the timeout sounds good to me, but we need to make sure to properly handle the case where retries are turn off/the task fails `max_retries` times. Assuming it's the behavior we want, need to make sure that the server's sequence number is increased to "skip" the failed task.

Reply:
Hmm yeah that's a good point about retries taking care of the issue. I think it should just work even if the actor has `max_reconstructs=0` since the receiving end deduplicates retried tasks.

Also, @edoakes, I think that issue about "skipping" failed tasks is already handled because of the `client_processed_up_to` field.